### PR TITLE
feat(FEC-10290): upgrade NPM packages 

### PR DESCRIPTION
**Pull request by dan-ziv**
### Description of the Changes

:white_check_mark: eslint upgraded to 7.x.x
:white_check_mark: babel upgraded to @babel (7.x.x)
:white_check_mark: webpack upgraded to 4.x.x

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -76,5 +72,8 @@ module.exports = {
       amd: 'shaka-player',
       root: 'shaka'
     }
+  },
+  optimization: {
+    minimize: PROD
```
Comment (OrenMe) [inline webpack.config.js:77]: use `--mode production`/`--mode development` in package.json instead(for build/dev/watch scripts)

File: .eslintrc.json [hunk 2]
```diff
@@ -13,7 +13,9 @@
     "should": true,
     "sinon": true,
     "__VERSION__": true,
-    "__NAME__": true
+    "__NAME__": true,
+    "process": true,
```
Comment (Yuvalke) [inline .eslintrc.json:17]: what is that added for?

Comment (dan-ziv) [reply .eslintrc.json:17]: `karma.config` and `webpack.config` was removed from `.eslintignore` (eslint should lint those files too...) and they are using Node globals that you need to define to eslint so it won't detect them as errors.

File: webpack.config.js (hunk 1 above)
Comment (dan-ziv) [reply webpack.config.js]: done

**Pull request merged**